sklearn/metrics/classification.py

```diff
@@ -2007,3 +2007,148 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     y_true = np.array(y_true == pos_label, int)
     y_true = _check_binary_probabilistic_predictions(y_true, y_prob)
     return np.average((y_true - y_prob) ** 2, weights=sample_weight)
+
+
+def calibration_loss(y_true, y_prob, sample_weight=None, reducer="sum",
```

Comment:
is brier_score a subcase of what calibration_loss can do? if so the code should be factorized to brier_score calls calibration_loss with the correct options.

Reply:
So to conclude on the experiments in this [gist](https://gist.github.com/samronsin/7177d1061f1030166dde754322b39e6c), I'd say that defaults that are close to the calibration curve (`sliding_window=False` and `bin_size_ratio=0.1`) would be good both in terms of metrics quality (measures accurately calibration) and consistency between loss and curve.

Reply:
Also Brier score is not quite a a subcase of any of these metrics (this would require to square the differences with `sliding_window=True` and `bin_size_ratio=1./N`).